src/quickgui/qgsquickutils.cpp

```diff
@@ -95,6 +98,20 @@ QgsPointXY QgsQuickUtils::transformPoint(QgsCoordinateReferenceSystem srcCrs, Qg
     return pt;
 }
 
+double QgsQuickUtils::distanceToMapUnits(QgsQuickMapSettings* mapSettings, double x1, double y1, double x2, double y2)
+{
+    QPointF p1 = mapSettings->coordinateToScreen(QgsPoint(x1, y1));
+    QPointF p2 = mapSettings->coordinateToScreen(QgsPoint(x2, y2));
+
+    QgsDistanceArea* mDistanceArea = new QgsDistanceArea();
```

Comment:
possible memory leak in future (imagine e.g. setEllipsoid raise C++ exception) use automatic instance [on stack](https://stackoverflow.com/a/27125830/2838364).